watcher/lib/influxdb/rollups.go

```diff
@@ -78,65 +78,16 @@ type Function struct {
 
 // Check verifies the function configuration is correct
 func (f Function) Check() error {
-	if !utils.OneOf(f.Function, constants.AllFunctions) &&
-		!strings.HasPrefix(f.Function, constants.FunctionPercentile) {
+	if !utils.OneOf(f.Function, constants.SimpleFunctions) && !isCompositeFunc(f) {
 		return trace.BadParameter(
-			"invalid Function, must be one of: %v, or start with %q",
-			constants.AllFunctions, constants.FunctionPercentile)
+			"invalid Function, must be one of: %v, or %v", constants.SimpleFunctions, constants.CompositeFunctions)
```

Comment:
I think the error message could be a little better, e.g. noting that composite functions require a parameter (your message gives an impression that it can be equal to a composite function prefix), e.g.:

```go
return trace.BadParameter("invalid Function, must be one of %v, or a composite function with one of %v prefixes",
    constants.SimpleFunctions, constants.CompositeFunctions)
```

Reply:
good idea, fixed.